dind/Dockerfile

```diff
@@ -1,26 +1,35 @@
 ARG DOCKER_VERSION
 
-FROM docker:${DOCKER_VERSION}-dind-rootless
+FROM golang:alpine AS build
+WORKDIR /build
+COPY docker-iptables-glue/go.* ./
+RUN go mod download
+COPY docker-iptables-glue/ ./
+RUN CGO_ENABLED=0 GOOS=linux go build
 
+FROM docker:${DOCKER_VERSION}-dind-rootless
 USER root
 
-RUN export CRUN_LATEST=$(wget -qO- https://api.github.com/repos/containers/crun/releases/latest | sed -En 's/.*"tag_name"\s*:\s*"((\\"|[^"])*)".*/\1/p') && \
-    wget -qO /usr/local/bin/crun "https://github.com/containers/crun/releases/download/${CRUN_LATEST}/crun-${CRUN_LATEST}-linux-amd64" && \
-    chmod +x /usr/local/bin/crun && \
-    export SLIRP4NETNS_LATEST=$(wget -qO- https://api.github.com/repos/rootless-containers/slirp4netns/releases/latest | sed -En 's/.*"tag_name"\s*:\s*"((\\"|[^"])*)".*/\1/p') && \
-    wget -qO /usr/local/bin/slirp4netns "https://github.com/rootless-containers/slirp4netns/releases/download/${SLIRP4NETNS_LATEST}/slirp4netns-x86_64" && \
-    chmod +x /usr/local/bin/slirp4netns && \
-    mkdir /opt/containerd && \
-    chown rootless:rootless /opt/containerd && \
-    mkdir /run/netns && \
-    chown rootless:rootless /run/netns
+ENV CRUN_BASE_DOWNLOAD_PATH=https://github.com/containers/crun/releases/download \
+    SLIRP4NETNS_BASE_DOWNLOAD_PATH=https://github.com/rootless-containers/slirp4netns/releases/download
 
-COPY ./init /sbin/init
+RUN apk add --no-cache iproute2 socat jq && \
+    export CRUN_LATEST=$(wget -qO- https://api.github.com/repos/containers/crun/releases/latest | jq -r .tag_name) && \
+    wget -qO /usr/local/bin/crun "${CRUN_BASE_DOWNLOAD_PATH}/${CRUN_LATEST}/crun-${CRUN_LATEST}-linux-amd64" && \
+    export SLIRP4NETNS_LATEST=$(wget -qO- https://api.github.com/repos/rootless-containers/slirp4netns/releases/latest | jq -r .tag_name) && \
+    wget -qO /usr/local/bin/slirp4netns "${SLIRP4NETNS_BASE_DOWNLOAD_PATH}/${SLIRP4NETNS_LATEST}/slirp4netns-x86_64" && \
+    chmod +x /usr/local/bin/crun /usr/local/bin/slirp4netns && \
+    mkdir /opt/containerd /run/netns && \
+    chown rootless:rootless /opt/containerd /run/netns
 
-USER rootless
+COPY ./init /sbin/
+COPY --from=build /build/docker-iptables-glue /sbin/
```

Comment:
I don't think that `/sbin` is the best place to have this file. We can leave it there for now, but I think we could merge `/sbin/int` with it and have a single binary instead of a script and a binary.

Reply:
Where would you suggest to put it?

Reply:
I'm not 100% sure. I would probably put it on `/usr/local/bin`, but it still doesn't seam right. Let's keep the way it is and I'll have `docker-iptables-glue` and `init.sh` merged.